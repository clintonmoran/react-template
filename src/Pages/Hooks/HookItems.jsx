import React from "react";

const HookItems = ({ items }) => {
  return (
    <ul data-testid="items">
      {items.map(item => {
        return <li data-testid="item" key={item.id}>{item.id}: {item.title}</li>;
      })}
    </ul>
  );
};

export default HookItems;
