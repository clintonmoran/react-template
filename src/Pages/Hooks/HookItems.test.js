import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render, cleanup, findByTestId, findAllByTestId } from '@testing-library/react'

import HookItems from './HookItems'

afterEach(cleanup);

const items = [
  { id: 1, abstract: "some abstract keys value", title: "KeysValue" },
  { id: 2, abstract: "2some abstract keys value", title: "2KeysValue" },
  { id: 3, abstract: "3some abstract keys value", title: "3KeysValue" },
  { id: 4, abstract: "4some abstract keys value", title: "4KeysValue" },
];

describe('<HookItems /> spec', () => {

  it('renders the component', () => {
    const { asFragment } = render(<HookItems items={items} />)
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a list of the correct length', async () => {
    const { findAllByTestId } = render(<HookItems items={items} />)
    const listItems = await findAllByTestId('item');
    expect(listItems).toHaveLength(4);
  });
});