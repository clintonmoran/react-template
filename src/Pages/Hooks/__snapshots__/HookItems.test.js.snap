// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HookItems /> spec renders the component 1`] = `
<DocumentFragment>
  <ul
    data-testid="items"
  >
    <li
      data-testid="item"
    >
      1: KeysValue
    </li>
    <li
      data-testid="item"
    >
      2: 2KeysValue
    </li>
    <li
      data-testid="item"
    >
      3: 3KeysValue
    </li>
    <li
      data-testid="item"
    >
      4: 4KeysValue
    </li>
  </ul>
</DocumentFragment>
`;
